common/docker/configuration.go

```diff
@@ -0,0 +1,335 @@
+package docker
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"pkg.world.dev/world-cli/common/config"
+)
+
+type GetDockerConfig func(cfg *config.Config) Config
+
+// DockerConfig is a configuration for a docker container
+// It contains the name of the container and a function to get the container and host config
+type Config struct {
+	Name string
+	*container.Config
```

Comment:
Is there a reason why this is a pointer? It's typically better to just have the value directly in struct composition to avoid nil deref issues.

Reply:
Just because function for creating container is accepting pointer parameters, but it can be handled by put `&` to get pointer of var passed. 

Changed this to the value, instead of using pointer to avoid nil